### ISSUE TITLE
ttnn.fill_implicit_tile_padding does not place the correct values when input is float32

### ISSUE THREAD

**Issue by nsorabaTT** [tenstorrent/tt-metal#22902]
## What is the behavior
`ttnn.fill_implicit_tile_padding` does not place the correct values in padded regions of a tile when input dtype is float32.

## Code Snippet
```python
import random
import torch
import ttnn

# Parameters
random.seed(0)
tensor_shape = [2, 2]

# Setup
device = ttnn.open_device(device_id=0)

torch_tensor = torch.tensor([-3.0, -5.0, -40, -9.0], dtype=torch.float32).reshape(tensor_shape)
ttnn_tensor = ttnn.from_torch(torch_tensor, layout=ttnn.TILE_LAYOUT, device=device)
ttnn_tensor = ttnn.fill_implicit_tile_padding(ttnn_tensor, fill_value=2)
ttnn_tensor = ttnn.reshape(ttnn_tensor, (32, 32), (32, 32))

ttnn_result_host = ttnn.from_device(ttnn_tensor)
ttnn_result = ttnn.to_torch(ttnn_result_host)
print(ttnn_result)

# Cleanup
ttnn.close_device(device)
```

## Why this matters
Reduction family of ops call implicit padding to pad +/-infinity as required for min/max. These produce incorrect results as a result of this bug.

**Comment by vsureshTT** [comment]
Issue seems to be sourced from
https://github.com/tenstorrent/tt-metal/blob/13e51419df31844609200828a0031c7aa76fe68e/ttnn/cpp/ttnn/operations/data_movement/fill_pad/device/fill_pad_program_factory.cpp

`tt::tt_metal::operation::ProgramWithCallbacks fill_pad_multi_core(const Tensor& input_tensor, float fill_value) {
`
`    // pack bf16 vals
    uint32_t packed_fill_value = (std::uint32_t)fill_value;
    if (input_tensor.get_dtype() == DataType::BFLOAT16) {
        packed_fill_value = pack_two_bfloat16_into_uint32({bfloat16(fill_value), bfloat16(fill_value)});
    } else if (input_tensor.get_dtype() == DataType::UINT16) {
        packed_fill_value = pack_two_uint16_into_uint32({fill_value, fill_value});
    }`

Since the fill value is passed as a float, it should be converted to a double before casting to uint32_t. 

@nsorabaTT added a change to address this in another PR of his for reduction, and should be able to fix it with that PR

**Comment by amorrisonTT** [comment]
I don't believe `fill_pad` supports float32, although this is not readily available in documentation
